node/src/reactor/main_reactor/reactor_state.rs

```diff
@@ -4,9 +4,58 @@ use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
 /// The state of the reactor.
+#[cfg_attr(doc, aquamarine::aquamarine)]
+/// ```mermaid
+/// flowchart TD
+///     style Start fill:#66ccff,stroke:#333,stroke-width:4px
+///     style End fill:#66ccff,stroke:#333,stroke-width:4px
+///     
+///     Start --> Initialize
+///     Initialize --> CatchUp
+///     CatchUp --> KeepUp
+///     KeepUp --> CatchUp
+///     KeepUp --> Validate
+///     Validate --> KeepUp
+///     CatchUp --> ShutdownForUpgrade
+///     KeepUp --> ShutdownForUpgrade
+///     Validate --> ShutdownForUpgrade
+///     CatchUp --> Upgrading
+///     CatchUp -->|at genesis| Validate
+///     Upgrading --> CatchUp
+///     ShutdownForUpgrade --> End
+/// ```
+/// ```mermaid
+/// flowchart TD
+///     style Start fill:#66ccff,stroke:#333,stroke-width:4px
+///     style End fill:#66ccff,stroke:#333,stroke-width:4px
+///     style F fill:#ffcc66,stroke:#333,stroke-width:4px
+///     style G fill:#ffcc66,stroke:#333,stroke-width:4px
+///     title[CatchUP process]
+///     title---Start
+///     style title fill:#FFF,stroke:#FFF
+///     linkStyle 0 stroke-width:0;
+///
+///     Start --> A["get sync identifier (sync starting point)"]
+///     A --> BlockHash
+///     A --> BlockIdentifier
+///     A --> SyncedBlockIdentifier
+///     A --> LocalTip
+///     BlockHash --> E[process identifier in<br/>block accumulator]
+///     BlockIdentifier --> E
+///     SyncedBlockIdentifier --> E
+///     LocalTip --> E
+///     CaughtUp --> H[handle upgrade<br/>if needed]
+///     H --> End
+///     E -->|more data needed<br/>from network| Leap
```

Comment:
I think we want to try and convey that we want the data to let us sync a block close to or at tip.  Something like:
"more data needed from network to let us sync near tip".  Not sure if that's better or not though!

Reply:
I added more info as suggested. It's ok and we also have plenty of place in this flowchart :)

09ca60ebb01c4151d3d19c75ec6b9715ca663d81